openpype/hosts/fusion/hooks/pre_fusion_profile_hook.py

```diff
@@ -152,10 +152,31 @@ def execute(self):
         self.log.info(f"Setting {fu_profile_dir_variable}: {fu_profile_dir}")
         self.launch_context.env[fu_profile_dir_variable] = str(fu_profile_dir)
 
+        self.set_masterprefs(profile_version)
+
+    def set_masterprefs(self, profile_version):
+        """Write the MasterPrefs file and set the environment variable"""
+
+        master_pref_contents = self.data["project_settings"]["fusion"].get(
+            "masterprefs", ""
+        )
+        if not master_pref_contents:
+            self.log.debug("No Fusion master preferences contents set in "
+                           "project settings. Skipping setting of Fusion "
+                           "master preferences.")
+            return
+
+        with tempfile.NamedTemporaryFile(
```

Comment:
Won't this spread preferences in temp folder? I mean, are they cleaned up in some way? Don't think it must be handled in this PR, but I can imagine that this could be an issue over time.

Reply:
> Won't this spread preferences in temp folder? I mean, are they cleaned up in some way? Don't think it must be handled in this PR, but I can imagine that this could be an issue over time.

Technically it would yes. It'd keep around the estimated 1 kb files for each launch of fusion until temporary folder is cleared by the user or the OS.

Reply:
Would it be possible to overwrite the file itself on each launch? That way we'll only have one file lingering in the temp folder.

Reply:
Is possible to have running more than one process of fusion? If yes, than that is not the way...

Reply:
> Would it be possible to overwrite the file itself on each launch? That way we'll only have one file lingering in the temp folder.

&

> Is possible to have running more than one process of fusion? If yes, then that is not the way...

Yes, you can run multiple instances of Fusion. I agree with @iLLiCiTiT that thus writing into a single file thus is not the way to go about this.